feat(curriculum): add Artifact Provenance Auditor workshop (#68860)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/workshop-artifact-provenance-auditor/6a57c331ae94fa4631b6086c.md

---
id: 6a57c331ae94fa4631b6086c
title: Step 13
challengeType: 1
dashedName: step-13
---

# --description--

> Artifacts go on and off display.

Reviewer: ```suggestion
Over time, artifacts frequently go on and off display.
```

> Create a function named `toggleDisplayStatus` with a parameter `id`. Look up the artifact with `collection[id]`. If the artifact exists, set its `onDisplay` property to the opposite of its current value.

Reviewer: ```suggestion
Create a `toggleDisplayStatus` function that takes an `id` parameter. Look up the artifact using `collection[id]`. If it exists, set its `onDisplay` property to the opposite of its current value.
```

# --hints--

You should have a function named `toggleDisplayStatus` with a parameter named `id`.

```js
const explorer = await __helpers.Explorer(code);
assert.exists(explorer.allFunctions.toggleDisplayStatus);
const params = explorer.allFunctions.toggleDisplayStatus.parameters;
assert.equal(params.length, 1);
assert.equal(params[0].toString(), "id");
```

Calling `toggleDisplayStatus` should flip the artifact's `onDisplay` value.

```js
const before = collection[101].onDisplay;
toggleDisplayStatus(101);

> assert.strictEqual(collection[101].onDisplay, !before);

Reviewer 2: to check that it is properly toggled, you should check with both a `before` of `true` and `before` of `false`, ie test both 101 and 102

---
id: 6a57c331ae94fa4631b6086c
title: Step 13
challengeType: 1
dashedName: step-13
---

# --description--

Artifacts go on and off display.

The lookup line is provided. Inside `toggleDisplayStatus`, if the artifact exists, set its `onDisplay` property to the opposite of its current value.

# --hints--

You should have a function named `toggleDisplayStatus` with a parameter named `id`.

```js
assert.isFunction(toggleDisplayStatus);
const regex = __helpers.functionRegex('toggleDisplayStatus', ['id']);

> assert.match(__helpers.removeJSComments(code), regex);

Reviewer: You can replace this with AST helpers. Please refer to the [AST Helpers documentation.](https://contribute.freecodecamp.org/curriculum-help/#typescript-helpers).
